workflow/controller/operator.go

```diff
@@ -262,6 +262,33 @@ func (woc *wfOperationCtx) operate() {
 		} else {
 			woc.globalParams[common.GlobalVarWorkflowStatus] = string(workflowStatus)
 		}
+
+		var failures []interface{}
+		for _, node := range woc.wf.Status.Nodes {
+			if node.Phase == wfv1.NodeFailed || node.Phase == wfv1.NodeError {
+				failures = append(failures, struct {
+					Name         string `json:"name"`
+					Message      string `json:"message"`
+					TemplateName string `json:"templateName"`
+					FinishedAt   string `json:"finishedAt"`
```

Comment:
these can be typed?

Reply:
Don't think that's super necessary, since this is an ephemeral struct

Reply:
change `finishedAt` to `time.Time` and put into a documented struct